* The last file's patch is thanks to Philipp Bunge in bug #189982
* This patch just makes OSX/Darwin/NeXT use .bundle as extension for
  dynamic libraries, as .so is an ugly thing on Darwin.

--- Python-2.5.1/configure.in
+++ Python-2.5.1/configure.in
@@ -1361,6 +1361,7 @@
 	case $ac_sys_system in
 	hp*|HP*)   SO=.sl;;
 	CYGWIN*)   SO=.dll;;
+	Darwin*)   SO=.bundle;;
 	*)	   SO=.so;;
 	esac
 else
--- Python-2.5.1/Python/dynload_next.c
+++ Python-2.5.1/Python/dynload_next.c
@@ -9,8 +9,8 @@
 #include <mach-o/dyld.h>
 
 const struct filedescr _PyImport_DynLoadFiletab[] = {
-	{".so", "rb", C_EXTENSION},
-	{"module.so", "rb", C_EXTENSION},
+	{".bundle", "rb", C_EXTENSION},
+	{"module.bundle", "rb", C_EXTENSION},
 	{0, 0}
 };
 
--- Python-2.5.1/Python/dynload_shlib.c
+++ Python-2.5.1/Python/dynload_shlib.c
@@ -46,11 +46,16 @@
         {"module.exe", "rb", C_EXTENSION},
         {"MODULE.EXE", "rb", C_EXTENSION},
 #else
+#ifdef __MACH__
+	{".bundle", "rb", C_EXTENSION},
+	{"module.bundle", "rb", C_EXTENSION},
+#else
 	{".so", "rb", C_EXTENSION},
 	{"module.so", "rb", C_EXTENSION},
 #endif
 #endif
 #endif
+#endif
 	{0, 0}
 };
 
